### PR TITLE
Check for a URL prefix and add it

### LINKED ISSUE #121: Jumping from WebUI->Settings or other way doesn't respect CSP URL Web Prefix
Trying out the new jump from WebUI -> Settings, or the other way around, there's a bug where the CSP Web URL Prefix is not being respected here.

Figured this out using the IIS Logs that hosts Web Gateway for us. We use a URL Prefix of /ucr on this instance and it is specified in the appropriate places e.g.,

`d $System.CSP.SetConfig("CSPConfigName","ucr")`objectscript

But when attempting to use the new feature, I can see in the IIS logs it isn't being thrown in:

`2022-02-07 18:20:45 W3SVC1 GET /ucr/isc/studio/usertemplates/gitsourcecontrol/webuidriver.csp/EGMYNS/MYPKG.Productions.Edge.CLS/api/settings  <-- Here I am in WebUI settings and then I click to jump to settings...

2022-02-07 18:20:45 W3SVC1 GET /isc/studio/usertemplates/gitsourcecontrol/gitprojectsettings.csp CSPSHARE=1&NSpace=EGMYNS&Username=MYUSERID <--- note no prefix`

I had to sanitize the log a bit but happy to show visually if it helps. Biggest issue here for me is that since it goes down a different url path, I lose my session credentials and it forces me to relogin! If I access Settings or WebUI from the Source Control menu directly, both still work fine. It's just the jump back and forth when in one or the other.

### PR DESCRIPTION
**Pull request by isc-svelury**
Checks the request URL for a prefix and prepends it to the base URL if it exists.

Closes #121

### CODE REVIEW

File: csp/gitprojectsettings.csp [hunk 1]
```diff
@@ -55,6 +55,15 @@ hr {
...
+
+ set locationOfAppPath = $LOCATE(%request.URL, appBasePath)
+
+ if (locationOfAppPath > 1) {
+    set beforeAppPath = $EXTRACT(%request.URL, 1, (locationOfAppPath-1))
+    set webuiURL = beforeAppPath_webuiURL
+ }
+     
  set settings = ##class(SourceControl.Git.Settings).%New()
  set ^test = settings
```
Comment (isc-tleavitt) [inline csp/gitprojectsettings.csp:68]: Oops - this looks like leftover debugging code? I know it wasn't added here, but let's remove it here.

File: csp/gitprojectsettings.csp [hunk 2]
```diff
@@ -55,6 +55,15 @@ hr {
  set $namespace = namespace
 
  set webuiURL = "/isc/studio/usertemplates/gitsourcecontrol/webuidriver.csp/"_$namespace_"/?CSPSHARE=1"
+ set appBasePath = "/isc/studio/usertemplates/gitsourcecontrol/"
```
Comment (isc-tleavitt) [inline csp/gitprojectsettings.csp:58]: Since this code is duplicated, I think it would make sense to refactor to a method somewhere.

**Review by isc-tleavitt — CHANGES_REQUESTED** [review]
Two minor changes I'd like to see.

File: csp/gitprojectsettings.csp (hunk 1 above)
Comment (isc-svelury) [reply csp/gitprojectsettings.csp:68]: Okay

File: csp/gitprojectsettings.csp (hunk 2 above)
Comment (isc-svelury) [reply csp/gitprojectsettings.csp:58]: Agreed. I tried it initially but I couldn't get it to work. This time, I haven't tried it. I think I have an idea though.

**Review by isc-tleavitt — APPROVED** [review]

**Pull request merged**